tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -46,11 +46,7 @@
     private Tracer() {}
 
     // Thread-safe since thread-local
-    private static final ThreadLocal<Trace> currentTrace = ThreadLocal.withInitial(() -> {
-        Trace trace = createTrace(Optional.empty(), Tracers.randomId());
-        MDC.put(Tracers.TRACE_ID_KEY, trace.getTraceId());
-        return trace;
-    });
+    private static final ThreadLocal<Trace> currentTrace = new ThreadLocal<>();
```

Comment:
OK so I guess this is the key change - this guy now uses `null` to represent the idea that we're not in a trace?

Reply:
That is correct, otherwise trace thread state leaks outside of root spans, which causes multiple root spans unless code directly creates a new traceId.